src/core/lib/promise/loop.h

```diff
@@ -203,7 +217,44 @@ class Loop {
 };
 
 template <typename F>
-Loop(F) -> Loop<F>;
+Loop(F) -> Loop<F, false>;
+
+// A version of Loop that yields the activity to another promise once per
+// iteration.
+template <typename F>
+auto YieldingLoop(F f) {
+  return Loop<F, true>(std::move(f));
+}
+
+template <typename N, typename PromiseFactory>
```

Comment:
Can it just be template <size_t N, typename PromiseFactory>

Reply:
Yes, but possibly trickier to use?